src/main.tsx

```diff
@@ -1,10 +1,44 @@
 import ReactDOM from 'react-dom/client';
+import { createBrowserRouter, RouterProvider } from 'react-router-dom';
+import SettingsPage from './pages/SettingsPage';
+import ErrorPage from './pages/ErrorPage';
+import QuizPage from './pages/QuizPage';
+import StatisticsPage from './pages/StatisticsPage';
+import ResultPage from './pages/ResultPage';
+import './main.css';
 
-import React from 'react';
-import App from './App.tsx';
+const router = createBrowserRouter([
+  {
+    path: '/',
+    element: <SettingsPage />,
+    errorElement: <ErrorPage />,
+  },
+  {
+    path: '/quiz',
+    element: <QuizPage />,
+    errorElement: <ErrorPage />,
+  },
+  {
+    path: '/result',
+    element: (
+      <ResultPage trueAnswer={0} totalAnswer={0} timeResult={''} category={''} difficulty={''} type={''} time={''} />
+    ),
+    errorElement: <ErrorPage />,
```

Comment:
have you checked if you could use ErrorPage for the path `*`
then I assume you may not put `errorElement` to the each route